azkaban-common/src/test/java/azkaban/ramppolicy/SimpleRampPolicyTest.java

```diff
@@ -0,0 +1,224 @@
+    RampPolicy policy = new SimpleRampPolicy(sysProps, privateProps);
+
+    Project project = new Project(1, PROJECT_ID);
+    Flow flow = new Flow(FLOW_ID);
+    ExecutableFlow executableFlow = new ExecutableFlow(project, flow);
+    ExecutableRamp executableRamp = ExecutableRamp.createInstance(
+        RAMP_ID,
+        RAMP_POLICY,
+        4, 3, false,
+        timeStamp, 0, timeStamp,
+        6, 2, 0, 4,
+        false, 100, false
+    );
+    Assert.assertFalse(policy.check(executableFlow, executableRamp));
+  }
+
+  @Test
+  public void Test_PausedRamp() throws Exception {
+    Props sysProps = new Props();
+    Props privateProps = new Props();
+    RampPolicy policy = new SimpleRampPolicy(sysProps, privateProps);
+
+    Project project = new Project(1, PROJECT_ID);
+    Flow flow = new Flow(FLOW_ID);
+    ExecutableFlow executableFlow = new ExecutableFlow(project, flow);
+    ExecutableRamp executableRamp = ExecutableRamp.createInstance(
+        RAMP_ID,
+        RAMP_POLICY,
+        4, 3, false,
+        timeStamp, 0, timeStamp,
+        6, 2, 0, 4,
+        true, 100, true
+    );
+    Assert.assertFalse(policy.check(executableFlow, executableRamp));
+  }
+
+  @Test
+  public void Test_ZeroPercentRamp() throws Exception {
+    Props sysProps = new Props();
+    Props privateProps = new Props();
+    RampPolicy policy = new SimpleRampPolicy(sysProps, privateProps);
+
+    Project project = new Project(1, PROJECT_ID);
+    Flow flow = new Flow(FLOW_ID);
+    ExecutableFlow executableFlow = new ExecutableFlow(project, flow);
+    ExecutableRamp executableRamp = ExecutableRamp.createInstance(
+        RAMP_ID,
+        RAMP_POLICY,
+        4, 3, false,
+        timeStamp, 0, timeStamp,
+        6, 2, 0, 4,
+        false, 0, true
+    );
+    Assert.assertFalse(policy.check(executableFlow, executableRamp));
+  }
+
+  @Test
+  public void Test_FullRamp() throws Exception {
+    Props sysProps = new Props();
+    Props privateProps = new Props();
+    RampPolicy policy = new SimpleRampPolicy(sysProps, privateProps);
+
+    Project project = new Project(1, PROJECT_ID);
+    Flow flow = new Flow(FLOW_ID);
+    ExecutableFlow executableFlow = new ExecutableFlow(project, flow);
+    ExecutableRamp executableRamp = ExecutableRamp.createInstance(
+        RAMP_ID,
+        RAMP_POLICY,
+        4, 3, false,
+        timeStamp, 0, timeStamp,
+        6, 2, 0, 4,
+        false, 100, true
+    );
+    Assert.assertTrue(policy.check(executableFlow, executableRamp));
+  }
+
+  @Test
+  public void Test_NegativePercentRamp() throws Exception {
+    Props sysProps = new Props();
+    Props privateProps = new Props();
+    RampPolicy policy = new SimpleRampPolicy(sysProps, privateProps);
+
+    Project project = new Project(1, PROJECT_ID);
+    Flow flow = new Flow(FLOW_ID);
+    ExecutableFlow executableFlow = new ExecutableFlow(project, flow);
+    ExecutableRamp executableRamp = ExecutableRamp.createInstance(
```

Comment:
Can you please consider using builder pattern for ExecutableRamp rather than createInstance?

Reply:
Agreed to address it in a seperate PR.
https://jira01.corp.linkedin.com:8443/browse/LIHADOOP-53102